minemeld/flask/sns.py

```diff
@@ -0,0 +1,105 @@
+import requests
+import json
+import os.path
+import uuid
+import minemeld
+from .logger import LOG
+from . import config
+
+SNSAPIURL = config.get('SNS_URL', 'https://o11kiw0vpe.execute-api.us-east-1.amazonaws.com/run')
+SNSENABLED = config.get('SNS_ENABLED', False)
+UUIDFILENAME = config.get('UUID_FILE', 'uu.id4')
+TYPEHELLO = 'hello'
+TYPEMKWISH = 'mkwish'
+TYPESTATS = 'stats'
+
+sns_obj = {}
+sns_available = False
+
+
+class sns:
+    def __init__(self, path):
+        self.api_url = SNSAPIURL
+        self.filename = os.path.join(path, UUIDFILENAME)
+        self.mm_version = minemeld.__version__
+        self.init_ok = self._init_uuid()
+
+    def get_status(self):
+        return self.init_ok
+
+    def _init_uuid(self):
+        uuid_error = uuid.UUID(bytes='\x01' * 16).hex
+        # Test case 1: UUIDFILENAME file does not exist. We try to create a new uuid and store in the filesystem
+        if not os.path.isfile(self.filename):
+            self.uuid = uuid.uuid4().hex
+            # If we've failed to send the one-in-a-lifetime hello we just don't store the uuid to try again next boot
+            if self._hello_world():
+                try:
+                    with open(self.filename, 'w') as f:
+                        f.write(self.uuid)
+                        LOG.debug('New uuid file created.')
+                except Exception as e:
+                    # Let the caller know uuid was not saved meaning sns might not be ready
+                    LOG.debug('Something went wrong creating the uuid file: {}'.format(self.filename))
+                except Exception:
```

Comment:
Duplicate `except` statement (see L41). Only the first handler will be called, L45 will never be executed

Reply:
OK